Functions for sample variance and sample standard deviation

wagafo opened the pull request:
I'm contributing simple functions for the sample standard deviation and variance of grouped data objects, just in case it can be considered to add these functions to this package. In case it, I'd also like to contribute functions for the skewness coefficient and the kurtosis in grouped data objects.

vigou3 commented:
Thanks Walter for these very well crafted contributions (and sorry for the long delay replying).

Would you just reorganize stuff as follows: combine `R/mean.grouped.data.R`, `R/var.grouped.data.R` and `R/sd.grouped.data.R` into, say, `R/summaries.grouped.data.R` and do the same for the `.Rd` files. This will keep the volume of files down, especially on the man pages front.

You could take the opportunity to add the other quantities you suggest.

Also, in the man page, perhaps include one or two examples showing that using, say, `var(x)` is equivalent to `emm(x, 2) - emm(x)^2`.

wagafo commented:
Thanks Vincent, I think it is a good idea to have a generic function "summaries" with all these measures. I will generate a new pull request for revision when I have it ready.

vigou3 commented:
Just to be clear: I was not referring to a single **function**, but rather to a single **file** where all functions are defined (and a single man page to document them).

wagafo commented:
Yes, thanks, it's clear. It may take a while but I will come back with a new proposal. By the way, please delete the other pull request I did in the cran repository for actuar, no hurry.

wagafo commented:
I have come back to to these functions. The two functions I implemented, are usually called "descriptive summaries for grouped data", see for instance Heumann et. at., "Introduction to Statistics and Data Analysis", page 53. The first raw moment and the "mean for grouped data" coincide, but for the second moment, the "variance for grouped data" underestimates the total variance because it only takes into account the variance **between** intervals and not the variance **within** intervals. I've been trying to relate emm(x,2) - emm(x)^2 with the "variance for grouped data", which is the functions I implemented, but I haven't been able to get an expression relating them. The first expression is always larger than the second, so I think (haven't been able to prove) that  emm(x,2) - emm(x)^2 is estimating the total variance (**between** and **within**), while my function is estimating only the variance **between** intervals. 
Do you think it is still worth it implementing these functions? I proposed them because they correspond to the descriptive summaries suggested by most textbooks for grouped data.

vigou3 commented:
Thanks Walter for your work on this. I think what is you propose is quite reasonable. There should be an explanation about the difference between `var(x)` and `emm(x, 2) - emm(x)^2` in `?var.grouped.data`, though.

Otherwise, there would be two main changes to make in your contribution before I can merge it:

1. arrange the code and the man pages in one file (well, one for each type) as mentioned last year;
2. it'd be a nicer interface to have `sd.grouped.data` and `var.grouped.data` called as methods of `sd` and `var`. Now, these two functions of **stats** are not generics, but we can define generics in the package; see section [7.1 of Writing R Extensions](https://cran.r-project.org/doc/manuals/R-exts.html#Adding-new-generics).

I can take care of item 2 if you're not comfortable doing it.
